pkg/apis/clusteroperator/validation/clusterversion.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/cluster-operator/pkg/apis/clusteroperator"
+)
+
+// ValidateClusterVersionName validates the name of a cluster version.
+var ValidateClusterVersionName = apivalidation.ValidateClusterName
+
+// ValidateClusterVersion validates a cluster version being created.
+func ValidateClusterVersion(cv *clusteroperator.ClusterVersion) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	for _, msg := range ValidateClusterVersionName(cv.Name, false) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("metadata").Child("name"), cv.Name, msg))
+	}
+
+	allErrs = append(allErrs, ValidateClusterVersionSpec(&cv.Spec, field.NewPath("spec"))...)
+
+	return allErrs
+}
+
+// ValidateClusterVersionStatusUpdate validates an update to the status of a clusterversion.
+func ValidateClusterVersionStatusUpdate(new *clusteroperator.ClusterVersion, old *clusteroperator.ClusterVersion) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// No validation required yet as the status struct is currently just a placeholder.
+
+	return allErrs
+}
+
+func ValidateClusterVersionSpec(spec *clusteroperator.ClusterVersionSpec, fldPath *field.Path) field.ErrorList {
```

Comment:
Add validation that at least one of the fields in `VMImages` is non-nil.

Reply:
It's probably going to be all nil in an on-premise use case, but we can validate that it's there for now, might save someone some time.

Reply:
If it's going to be all nil, then the `VMImages` field itself should probably be optional. I would find it frustrating as a user to be forced to supply a `VMImages` field with nothing in it.